modules/c++/sys/source/OSUnix.cpp

```diff
@@ -56,7 +56,7 @@ std::string readLink(const std::string& pathname)
 
     if (readlink(pathname.c_str(), buffer, PATH_MAX) == -1)
     {
-        throw except::Exception(Ctxt(strerror(errno)));
+        return "";
     }
```

Comment:
I've changed it so it checks if the path exists before calling readlink. The issue I was encountering was that on Solaris the symlink is just elsewhere. Though a StackOverflow user did say that his system required root to follow the symlink in `/proc/self/exe`.